### PR TITLE
Bugfix/65 improper handling of download url parameter

### LINKED ISSUE #65: Improper handling of download_url parameter
`download_url` parameter in main class `jboss` has been moved to params file in commit https://github.com/coi-gov-pl/puppet-jboss/commit/26c7345022ab95b619c95cfabdb3720acce6b672 (PR #26). This causes numerous issues when overriding `$product` or `$version` parameters using main class. Like this:

```puppet
class { 'jboss':
  product => 'wildfly',
  version => '9.0.2.Final',
}
```

In this example `$download_url` is being resolved in `params.pp` file and default values of `$product` and `$version` are being used.

There are 2 propositions to fix this:

#### Move download_url to runtime class and calculate there

* Moving `$download_url` to `jboss::internal::runtime` class
* Set parameters from `jboss` class to `jboss::internal::runtime` class
* Use parameters from `jboss::internal::runtime` class instead of `params.pp`
* Update all references to `$jboss::internal::runtime::download_url` instead of `$jboss::params::download_url`

#### Calculate in params using product and version in runtime class

* Set parameters from `jboss` class to `jboss::internal::runtime` class
* Use parameters from `jboss::internal::runtime` class instead of `params.pp`

### PR DESCRIPTION
**Pull request by Cosaquee**
fixes #65

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/coi-gov-pl/puppet-jboss/pull/80" target="_blank">ReviewNinja

**Comment by Cosaquee** [comment]
@coi-gov-pl/jboss-puppet-team please review

**Comment by therek** [comment]
:+1:

**Comment by therek** [comment]
:+1:

File: manifests/internal/package.pp [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Internal class that installs JBoss
 class jboss::internal::package (
-  $download_url,
+  $download_url     = $jboss::internal::runtime::download_url,
```
Comment (cardil) [inline manifests/internal/package.pp]: Puppetlabs convention

File: spec/classes/init_spec.rb [hunk 2]
```diff
@@ -10,7 +10,6 @@
       is_expected.to contain_class('jboss').with({
         :product      => 'wildfly',
         :version      => '9.0.2.Final',
-        :download_url => 'http://download.jboss.org/wildfly/9.0.2.Final/wildfly-9.0.2.Final.zip'
         })
```
Comment (cardil) [inline spec/classes/init_spec.rb]: Check is `undef`, and valid path in `jboss::internal::package`

**Comment by cardil** [comment]
:+1: conditional one :smile_cat:

**Pull request merged**